libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -165,6 +167,26 @@ void AC_PosControl::set_alt_target_from_climb_rate(float climb_rate_cms, float d
         _pos_target.z += climb_rate_cms * dt;
     }
 
+#if APM_BUILD_TYPE(APM_BUILD_ArduSub)
+    // do not let target alt get above limit
+    if (_alt_max < 100 && _pos_target.z > _alt_max) {
+        _pos_target.z = _alt_max;
+        _limit.pos_up = true;
+    }
+
+    // do not let target alt get below limit
+    if (_alt_min < 0 && _alt_min < _alt_max && _pos_target.z < _alt_min) {
+        _pos_target.z = _alt_min;
+        _limit.pos_down = true;
+    }
+#else
+    // do not let target alt get above limit
+    if (_alt_max > 0 && _pos_target.z > _alt_max) {
+        _pos_target.z = _alt_max;
+        _limit.pos_up = true;
+    }
+#endif
```

Comment:
Alt max works the same way as in Copter, and we keep that at the surface of the water, so that the vehicle doesn't try to fly when it reaches the surface. Alt min (max depth), just prevents the vehicle from exceeding the set depth in the same manner.